src/sidebar/components/autocomplete-list.js

```diff
@@ -4,6 +4,23 @@ import { useMemo } from 'preact/hooks';
 import propTypes from 'prop-types';
 
 const defaultListFormatter = item => item;
+/**
+ * @typedef AutocompleteListProps
+ * @prop {number} [activeItem] - The index of the highlighted item.
+ * @prop {string} [id] - Optional unique HTML attribute id. This can be used
+ *   for parent `aria-controls` coupling.
+ * @prop {string} [itemPrefixId] - Optional unique HTML attribute id prefix
+ *   for each item in the list. The final value of each items' id is
+ *   `{itemPrefixId}{activeItem}`
+ * @prop {(Object|string)[]} list - The list of items to render. This can be a simple
```

Comment:
If you wanted to make the connection between `list` and `listFormatter` more obvious you could use a template parameter for the object type rather than `Object`. I'd only do that if it is easy and doesn't surface a bunch of extra complications though.

Reply:
I think I know what you mean, and that seems reasonable. I'll give it a go